src/braket/devices/device.py

```diff
@@ -104,3 +110,43 @@ def status(self) -> str:
             str: The status of this Device
         """
         return self._status
+
+    def _validate_device_noise_model_support(self) -> None:
+        noise_pragma_name_to_class = copy(one_prob_noise_map)
+        noise_pragma_name_to_class.update({"kraus": Noise.Kraus})
+
+        supported_noises = [
+            noise_pragma_name_to_class[pragma.split("braket_noise_")[-1]]
+            for pragma in (self.properties.action[DeviceActionType.OPENQASM].supportedPragmas)
+            if "braket_noise_" in pragma
+        ]
+
+        noise_operators = [noise_instr.noise for noise_instr in self._noise_model._instructions]
```

Comment:
How about instead of using `supportedPragmas`, using `supportedOperations`? The names are normalized there. Then you can do something like
```python
supported_noise = set(action.supportedOperations).intersection(set(noise_pragma_name_to_class.keys()))
```

Reply:
I've considered that but the noise is in supportedOperations for local simulator, but in supportedPragmas in AwsDevice.

Reply:
Oof, in that case, as verbose as it is, just create the map from the properties pragma name "braket_pragma_..." to the actual name. It's ugly, but at least it's explicit instead of relying on string parsing.

Reply:
I see you you meant now. I added translation from pragma to noise class. Even though the validation only compares the sets of names. I set translation to map to noise class because I think it's more robust than mapping to names.